tests/test_transform.py

```diff
@@ -668,13 +668,104 @@ def checkGetJacobian(self, fromName, toName):
+            Point2Endpoint) for the final, concatenated Transform
+        midName : string
+            the prefix for the shared endpoint where two Transforms will be
+            concatenated
+        toName : string
+            the prefix of the ending endpoint for the final, concatenated
+            Transform
+        """
+        transform1Class = getattr(afwGeom,
+                                  "Transform{}To{}".format(fromName, midName))
+        transform2Class = getattr(afwGeom,
+                                  "Transform{}To{}".format(midName, toName))
+        baseMsg = "{}.of({})".format(transform2Class.__name__,
+                                     transform1Class.__name__)
+        for nIn in self.goodNaxes[fromName]:
+            for nMid in self.goodNaxes[midName]:
+                for nOut in self.goodNaxes[toName]:
+                    msg = "{}, nIn={}, nMid={}, nOut={}".format(
+                        baseMsg, nIn, nMid, nOut)
+                    polyMap = makeTwoWayPolyMap(nIn, nMid)
+                    transform1 = transform1Class(polyMap)
+                    polyMap = makeTwoWayPolyMap(nMid, nOut)
+                    transform2 = transform2Class(polyMap)
+                    transform = transform2.of(transform1)
+
+                    fromEndpoint = transform1.getFromEndpoint()
+                    toEndpoint = transform2.getToEndpoint()
+
+                    inPoint = fromEndpoint.pointFromData(makeRawPointData(nIn))
+                    outPointMerged = transform.tranForward(inPoint)
+                    outPointSeparate = transform2.tranForward(
+                        transform1.tranForward(inPoint))
+                    assert_allclose(toEndpoint.dataFromPoint(outPointMerged),
+                                    toEndpoint.dataFromPoint(outPointSeparate),
+                                    err_msg=msg)
+
+                    outPoint = toEndpoint.pointFromData(makeRawPointData(nOut))
+                    inPointMerged = transform.tranInverse(outPoint)
+                    inPointSeparate = transform1.tranInverse(
+                        transform2.tranInverse(outPoint))
+                    assert_allclose(
+                        fromEndpoint.dataFromPoint(inPointMerged),
+                        fromEndpoint.dataFromPoint(inPointSeparate),
+                        err_msg=msg)
+
+        with self.assertRaises(InvalidParameterError):
+            polyMap = makeTwoWayPolyMap(2, 3)
+            transform1 = transform1Class(polyMap)
+            transform2 = transform2Class(polyMap)
+            transform = transform2.of(transform1)
+
+        if fromName != midName:
+            with self.assertRaises(InvalidParameterError):
+                polyMap = makeTwoWayPolyMap(2, 3)
+                transform1 = transform1Class(polyMap)
+                polyMap = makeTwoWayPolyMap(3, 3)
+                transform2 = transform1Class(polyMap)
+                transform = transform2.of(transform1)
```

Comment:
If you are testing type mismatch then I think the main thing is to test at least one set of valid axes for a given endpoint type. Also, the #s of axes should match if that is allowed for the given types, so you know you are testing type mismatch instead of # of axes mismatch.

However, may be easier and more thorough to cycle through all valid #s of axes for each endpoint type. One could imagine that code testing both type mismatch and axis mismatch, e.g. cycle through all choices and if types differ or #s of axes differ then assert that `of` will throw an exception.